src/operator/controllers/intents_reconcilers/networkpolicy/reconciler.go

```diff
@@ -474,6 +475,45 @@ func (r *Reconciler) reconcileEndpointsForPolicy(ctx context.Context, newPolicy
 	return r.extNetpolHandler.HandlePodsByLabelSelector(ctx, newPolicy.Namespace, selector)
 }
 
+func (r *Reconciler) handleExistingPolicyRetry(ctx context.Context, ep effectivepolicy.ServiceEffectivePolicy, netpol *v1.NetworkPolicy) error {
+	existingPolicy := &v1.NetworkPolicy{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      netpol.Name,
+		Namespace: ep.Service.Namespace},
+		existingPolicy)
+	if err != nil && !k8serrors.IsNotFound(err) {
+		r.RecordWarningEventf(existingPolicy, consts.ReasonGettingNetworkPolicyFailed, "failed to get network policy: %s", err.Error())
+		return errors.Wrap(err)
+	}
+	if k8serrors.IsNotFound(err) {
+		return errors.Wrap(fmt.Errorf("failed creating network policy with AlreadyExists err, but failed getting netpol from k8s api server"))
+	}
+
+	return r.updateExistingPolicy(ctx, ep, existingPolicy, netpol)
+}
+
+func (r *Reconciler) handleCreationErrors(ctx context.Context, ep effectivepolicy.ServiceEffectivePolicy, netpol *v1.NetworkPolicy, err error) error {
+	errStr := err.Error()
+	if k8serrors.IsAlreadyExists(err) {
+		// Ideally we would just return {Requeue: true} but it is not possible without a mini-refactor
+		return r.handleExistingPolicyRetry(ctx, ep, netpol)
+	}
+
+	if k8serrors.IsForbidden(err) && strings.Contains(errStr, "is being terminated") {
+		// Namespace is being deleted, nothing to do further
+		logrus.Debugf("Namespace %s is being terminated, ignoring api server error", netpol.Namespace)
+		return nil
+	}
+
+	if strings.Contains(errStr, netpol.Namespace) && strings.Contains(errStr, "not found") {
```

Comment:
should we check the error type here? (should be NotFound error)

Reply:
Wasn't sure if for missing namespace on Create that returning error type is `NotFound`, but it is. Changed :+1: